src/common/lib/python/common_extension.cc

```diff
@@ -387,78 +368,25 @@ static int PyTask_init(PyTask *self, PyObject *args, PyObject *kwds) {
     required_resources["CPU"] = 1.0;
   }
 
-  Py_ssize_t num_args = PyList_Size(arguments);
-
-  bool use_raylet = false;
-  if (use_raylet_object != nullptr && PyObject_IsTrue(use_raylet_object) == 1) {
-    use_raylet = true;
-  }
-  self->spec = nullptr;
-  self->task_spec = nullptr;
-
+  std::function<PyObject *(PyObject *)> callMethodObjArgs =
+    std::bind(PyObject_CallMethodObjArgs, pickle_module, pickle_dumps,
+                std::placeholders::_1, pickle_protocol, NULL);
+  std::function<bool(PyObject *)> isPyObjectIDType = 
+    std::bind(PyObject_IsInstance, std::placeholders::_1,
+      reinterpret_cast<PyObject *>(&PyObjectIDType));
```

Comment:
`callMethodObjArgs ` and `isPyObjectIDType` don't seem dynamic? why make it a parameter?

Reply:
To resolve the dependency, the PyObjectIDType and pickle_module is invisible for the common_task. By the function, the common_task don't need take dependency on it.